### ISSUE TITLE
NWN: Error starting the engine when using the Steam version

### ISSUE THREAD

**Issue by seedhartha** [xoreos/xoreos#552]
Apparently, the directory structure was changed in the Steam version of Neverwinter Nights. The engine fails to start with error message "Failed to index mandatory directory "data"" and subsequent segmentation fault.

**Comment by DrMcCoy** [comment]
Yeah, that the Steam version doesn't work is known. Probably needs a probe and some flag in the engine class, too, to properly distinguish it.

It shouldn't segfault, though, it should just quit with an error :/

**Comment by DrMcCoy** [comment]
The Steam version changes quite a lot, in fact. It's very possible that it needs way more work.

Also, I'm not sure if Beamdog is still changing things. That was basically the reason I didn't follow the xoreos' code with the changes back when the it was in beta.

So I guess the first steps are:
- Write a probe
- Extend the Version class to detect the Beamdog version, too
- Implement loading resource with the changed directory layout
- Find out if the Beamdog NWN version is stable now
- Check for further changes we need to heed

AFAIK, the videos are using a WebM container with VP8 and Vorbis in them. I had talked to @clone2727 about that back then, though I don't remember if we added all the necessary code to play them already.

**Comment by clone2727** [comment]
I need to finish the Matroska code for video, but the codec stuff is all done

**Comment by kblaschke** [comment]
[This line will segfault](https://github.com/xoreos/xoreos/blob/84cab32f77b66e5078d07d92e8e002cf5e1c8205/src/graphics/aurora/model_nwn.cpp#L1190) if the mesh has an unknown type and won't be loaded. The `_mesh` pointer isn't checked before [this line](https://github.com/xoreos/xoreos/blob/84cab32f77b66e5078d07d92e8e002cf5e1c8205/src/graphics/aurora/model_nwn.cpp#L1202).

The Beamdog version seems to have a new mesh type `animmesh`.

Instead of testing for specific types [here](https://github.com/xoreos/xoreos/blob/84cab32f77b66e5078d07d92e8e002cf5e1c8205/src/graphics/aurora/model_nwn.cpp#L1107), checking `_mesh` to be a valid pointer instead might be advisable.
